:include-table: breaks when a csv column header is named "title"

If a csv file has the word title as a column header, Znai throws an exception.

```
Caused by: java.lang.RuntimeException: error handling include plugin <table>
  free param: example.csv
  opts: {}

parameter <title> is already registered

    at org.testingisdocumenting.znai.parser.commonmark.MarkdownVisitor.handleIncludePlugin (MarkdownVisitor.java:221)
    at org.testingisdocumenting.znai.parser.commonmark.MarkdownVisitor.visit (MarkdownVisitor.java:145)
    at org.testingisdocumenting.znai.parser.commonmark.include.IncludeBlock.accept (IncludeBlock.java:43)
...
Caused by: java.lang.IllegalArgumentException: parameter <title> is already registered
    at org.testingisdocumenting.znai.extensions.PluginParamsDefinition.validateNameUniqueness (PluginParamsDefinition.java:98)
    at org.testingisdocumenting.znai.extensions.PluginParamsDefinition.add (PluginParamsDefinition.java:91)
    at org.testingisdocumenting.znai.extensions.PluginParamsDefinition.add (PluginParamsDefinition.java:71)
```

The csv file can use the word title, as long as it's a value, not a header.

`:include-table: example.csv`

```csv
header name
some words
title
value
```

--- a/znai-core/src/main/java/org/testingisdocumenting/znai/extensions/PluginParamsDefinition.java
+++ b/znai-core/src/main/java/org/testingisdocumenting/znai/extensions/PluginParamsDefinition.java
@@ -67,6 +67,10 @@ public class PluginParamsDefinition {
         this.isDefined = isDefined;
     }
 
+    public boolean has(String name) {
+        return params.stream().anyMatch(p -> p.name.equals(name));
+    }
+
     public PluginParamsDefinition add(String name, PluginParamType type, String description, String example) {
         add(new Param(name, type, description, example, false));
         return this;

--- a/znai-docs/znai/layout/table/table-conflict.csv
+++ b/znai-docs/znai/layout/table/table-conflict.csv
@@ -1,0 +1,3 @@
+account, title
+#12BGD3, custom table with a long attachment
+#12BGD4, custom table with a short attachment

--- a/znai-docs/znai/layout/tables.md
+++ b/znai-docs/znai/layout/tables.md
@@ -145,6 +145,14 @@ Note: total percentage of all columns can go above 100%, in which case a scroll 
 
 :include-table: table/table.csv {Price: {width: "50%"}, "Description": {width: "60%"}}
 
+# Column Names Conflict
+
+Use `_` prefix if a column name matches the plugin parameter name like `title` when setting column specific parameters.
+
+    :include-table: table/table-conflict.csv {"_title": {width: "80%"}}
+
+:include-table: table/table-conflict.csv {"_title": {width: "80%"}}
+
 # Min Width
 
 Use `minColumnWidth` to set min width for all table columns

--- a/znai-docs/znai/release-notes/1.63/fix-2023-02-08-table-params-conflict.md
+++ b/znai-docs/znai/release-notes/1.63/fix-2023-02-08-table-params-conflict.md
@@ -1,0 +1,1 @@
+* Fix: Table column parameters [name conflict](layout/tables#column-names-conflict)

--- a/znai-website-gen/src/main/java/org/testingisdocumenting/znai/extensions/table/TableDocElementFromParams.java
+++ b/znai-website-gen/src/main/java/org/testingisdocumenting/znai/extensions/table/TableDocElementFromParams.java
@@ -69,7 +69,11 @@ class TableDocElementFromParams {
         List<Map<String, Object>> columns = (List<Map<String, Object>>) tableAsMap.get("columns");
 
         opts.forEach((columnName, meta) -> {
-            Optional<Map<String, Object>> column = columns.stream().filter(c -> c.get("title").equals(columnName)).findFirst();
+            Optional<Map<String, Object>> column = columns.stream().filter(c -> {
+                Object columnTitle = c.get("title");
+                // we use _ for column names parameters that match plugin parameters (e.g. title)
+                return columnName.equals(columnTitle) || columnName.equals("_" + columnTitle);
+            }).findFirst();
             column.ifPresent(c -> c.putAll((Map<? extends String, ?>) meta));
         });
 

--- a/znai-website-gen/src/main/java/org/testingisdocumenting/znai/extensions/table/TableIncludePlugin.java
+++ b/znai-website-gen/src/main/java/org/testingisdocumenting/znai/extensions/table/TableIncludePlugin.java
@@ -66,7 +66,10 @@ public class TableIncludePlugin implements IncludePlugin {
                                 ParserHandler parserHandler,
                                 Path markupPath,
                                 PluginParams pluginParams) {
-        docElementFromParams = new TableDocElementFromParams(componentsRegistry, tableDataFromContentAndParams, markupPath, pluginParams, componentsRegistry.defaultParser(), fullPath);
+        docElementFromParams = new TableDocElementFromParams(componentsRegistry,
+                tableDataFromContentAndParams, markupPath, pluginParams,
+                componentsRegistry.defaultParser(),
+                fullPath);
         return docElementFromParams.create();
     }
 

--- a/znai-website-gen/src/main/java/org/testingisdocumenting/znai/extensions/table/TablePluginParams.java
+++ b/znai-website-gen/src/main/java/org/testingisdocumenting/znai/extensions/table/TablePluginParams.java
@@ -42,8 +42,18 @@ class TablePluginParams {
         PluginParamsDefinition result = new PluginParamsDefinition();
         result.add(commonParamsDefinition);
 
-        tableData.columnNamesStream().forEach(columnName ->
-                result.add(columnName, PluginParamType.OBJECT, "column <" + columnName + "> config", "{width: \"50%\"}"));
+        tableData.columnNamesStream().forEach(columnName -> {
+            String description = "column <" + columnName + "> config";
+            String example = "{width: \"50%\"}";
+
+            // for cases when column name matches existing parameters like "title",
+            // we ask users to use _title
+            if (result.has(columnName)) {
+                result.add("_" + columnName, PluginParamType.OBJECT, description, example);
+            } else {
+                result.add(columnName, PluginParamType.OBJECT, description, example);
+            }
+        });
 
         return result;
     }

--- a/znai-website-gen/src/test/groovy/org/testingisdocumenting/znai/extensions/table/TableIncludePluginTest.groovy
+++ b/znai-website-gen/src/test/groovy/org/testingisdocumenting/znai/extensions/table/TableIncludePluginTest.groovy
@@ -44,6 +44,16 @@ class TableIncludePluginTest {
     }
 
     @Test
+    void "should use underscore to distinct column names that match plugin parameter names"() {
+        def element = process('test-table-title-column.json {_title: {width: "30%"}}')
+
+        element.should == [table: [columns: [[title: "account"], [title: "title", width: "30%"]],
+                                   data:[[[[markup: "#12BGD3", type: "TestMarkup"]],
+                                          [[markup: "custom table with a long attachment", type: "TestMarkup"]]]]],
+                           type: "Table"]
+    }
+
+    @Test
     void "should filter out columns from json based on case insensitive provided names"() {
         def element = process('test-table.json', '{columns: ["account", "description"]}')
 

--- a/znai-website-gen/src/test/resources/test-table-title-column.json
+++ b/znai-website-gen/src/test/resources/test-table-title-column.json
@@ -1,0 +1,1 @@
+[{"account": "#12BGD3", "title": "custom table with a long attachment"}]

